src/components/enterprise-user-subsidy/data/utils.js

```diff
@@ -15,3 +18,49 @@ export const transformRedeemablePoliciesData = (policies) => policies?.map(polic
     learner_content_assignments: assignmentsWithSubsidyExpiration,
   };
 });
+
+/**
+ * Determine whether learner has only content assignments available to them, based on the presence of:
+ * - content assignments for display (allocated or canceled)
+ * - no auto-applied budgets
+ * - no current enterprise offers
+ * - no active license or license requests
+ * - no assigned codes or code requests
+ *
+ * @param {Object} params - The parameters object.
+ * @param {Object} params.subscriptionPlan - The subscription plan of the learner.
+ * @param {Object} params.subscriptionLicense - The subscription license of the learner.
+ * @param {Array} params.licenseRequests - The license requests of the learner.
+ * @param {number} params.couponCodesCount - The count of assigned coupon codes of the learner.
+ * @param {Array} params.couponCodeRequests - The coupon code requests of the learner.
+ * @param {Object} params.redeemableLearnerCreditPolicies - The redeemable learner credit policies.
+ * @param {boolean} params.hasCurrentEnterpriseOffers - Whether the learner has current enterprise offers.
+ * @returns {boolean} - Returns true if the learner has only content assignments available to them, false otherwise.
+ */
+export const determineLearnerHasContentAssignmentsOnly = ({
+  subscriptionPlan,
+  subscriptionLicense,
+  licenseRequests,
+  couponCodesCount,
+  couponCodeRequests,
+  redeemableLearnerCreditPolicies,
+  hasCurrentEnterpriseOffers,
+}) => {
+  const hasActiveLicense = !!(subscriptionPlan?.isActive && subscriptionLicense?.status === LICENSE_STATUS.ACTIVATED);
+  const hasActiveLicenseOrLicenseRequest = hasActiveLicense || licenseRequests.length > 0;
+  const hasAssignedCodesOrCodeRequests = couponCodesCount > 0 || couponCodeRequests.length > 0;
+  const hasAutoAppliedLearnerCreditPolicies = !!redeemableLearnerCreditPolicies?.redeemablePolicies.filter(
+    policy => policy.policyType !== POLICY_TYPES.ASSIGNED_CREDIT,
+  ).length > 0;
```

Comment:
Is it worth making this filter more explicit and have it pick only the redeemable policies whose type is `PER_LEARNER_CREDIT`?

Reply:
I opted for the current implementation so that if/when we do start using per-learner enrollment limits, no code changes would be needed here. That said, I agree, it'd be better to be explicit. For example, if a new policy type for BnR is created, we wouldn't that to be treated as assignable. Will update!